From de154cd65e16420abc726bd25b5a35c70b81623f Mon Sep 17 00:00:00 2001
Message-Id: <de154cd65e16420abc726bd25b5a35c70b81623f.1618500237.git.stefan@agner.ch>
In-Reply-To: <9c98c07abbe9ee675ec588ad7c15f0285e8a0f07.1618500237.git.stefan@agner.ch>
References: <9c98c07abbe9ee675ec588ad7c15f0285e8a0f07.1618500237.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Wed, 17 Mar 2021 14:48:43 +0100
Subject: [PATCH 5/9] package/linux-firmware: add RTL87XX/RTL88XX Bluetooth
 firmware

Add firmware files for Realtek 87XX and 88XX Bluetooth chipsets. Those
are supported by the upstream Realtek Bluetooth driver CONFIG_BT_RTL.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 package/linux-firmware/Config.in         | 10 ++++++++++
 package/linux-firmware/linux-firmware.mk | 21 +++++++++++++++++++++
 2 files changed, 31 insertions(+)

diff --git a/package/linux-firmware/Config.in b/package/linux-firmware/Config.in
index a41310ff2a..ba3322fa18 100644
--- a/package/linux-firmware/Config.in
+++ b/package/linux-firmware/Config.in
@@ -79,6 +79,16 @@ config BR2_PACKAGE_LINUX_FIRMWARE_AR3012_USB
 	help
 	  Firmware files for Atheros bluetooth chipset AR3012 on USB.
 
+config BR2_PACKAGE_LINUX_FIRMWARE_RTL_87XX_BT
+	bool "Realtek 87xx Bluetooth"
+	help
+	  Realtek 8723a/8723b/8723bs/8723d/8723ds/8761a
+
+config BR2_PACKAGE_LINUX_FIRMWARE_RTL_88XX_BT
+	bool "Realtek 88xx Bluetooth"
+	help
+	  Realtek 8821a/8821c/8822b/8822cs/8822cu/8852au
+
 config BR2_PACKAGE_LINUX_FIRMWARE_TI_CC2560
 	bool "TI CC2560"
 	help
diff --git a/package/linux-firmware/linux-firmware.mk b/package/linux-firmware/linux-firmware.mk
index f23da171c6..9c512ab750 100644
--- a/package/linux-firmware/linux-firmware.mk
+++ b/package/linux-firmware/linux-firmware.mk
@@ -64,6 +64,27 @@ LINUX_FIRMWARE_FILES += qca/rampatch_usb_00000302.bin qca/nvm_usb_00000302.bin
 LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENSE.qcom
 endif
 
+# Realtek 87xx Bluetooth
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_RTL_87XX_BT),y)
+LINUX_FIRMWARE_FILES += \
+	rtl_bt/rtl8723a_fw.bin rtl_bt/rtl8723b_fw.bin \
+	rtl_bt/rtl8723bs_config-OBDA8723.bin \
+	rtl_bt/rtl8723bs_fw.bin rtl_bt/rtl8723d_config.bin \
+	rtl_bt/rtl8723d_fw.bin rtl_bt/rtl8761a_fw.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.rtlwifi_firmware.txt
+endif
+
+# Realtek 88xx Bluetooth
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_RTL_88XX_BT),y)
+LINUX_FIRMWARE_FILES += \
+	rtl_bt/rtl8812ae_fw.bin rtl_bt/rtl8821a_fw.bin \
+	rtl_bt/rtl8821c_config.bin rtl_bt/rtl8821c_fw.bin \
+	rtl_bt/rtl8822b_config.bin rtl_bt/rtl8822b_fw.bin \
+	rtl_bt/rtl8822cs_config.bin rtl_bt/rtl8822cs_fw.bin \
+	rtl_bt/rtl8822cu_config.bin rtl_bt/rtl8822cu_fw.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.rtlwifi_firmware.txt
+endif
+
 # rt2501/rt61
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_RALINK_RT61),y)
 LINUX_FIRMWARE_FILES += rt2561.bin rt2561s.bin rt2661.bin
-- 
2.31.1

